File: SourceCode/Samples/Compass sensor sample/C#/Shared/SampleConfiguration.cs
﻿//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

using System;
using System.Collections.Generic;
using Windows.UI.Xaml.Controls;

namespace SDKTemplate
{
    public partial class MainPage : Page
    {
        public const string FEATURE_NAME = "Compass";

        List<Scenario> scenarios = new List<Scenario>
        {
            new Scenario() { Title = "Data Events", ClassType = typeof(Microsoft.Samples.Devices.Sensors.CompassSample.Scenario1) },
            new Scenario() { Title = "Polling", ClassType = typeof(Microsoft.Samples.Devices.Sensors.CompassSample.Scenario2) },
            new Scenario() { Title = "Calibration", ClassType = typeof(Microsoft.Samples.Devices.Sensors.CompassSample.Scenario3) }
        };
    }

    public class Scenario
    {
        public string Title { get; set; }

        public Type ClassType { get; set; }

        public override string ToString()
        {
            return Title;
        }
    }
}

File: SourceCode/Samples/App tiles and badges sample/C# and C++/Shared/SampleConfiguration.cpp
﻿//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "MainPage.xaml.h"
#include "SampleConfiguration.h"

using namespace SDKSample;

using namespace Windows::UI::ViewManagement;

Platform::Array<Scenario>^ MainPage::scenariosInner = ref new Platform::Array<Scenario>
{
    { "Send tile notification with text", "SDKSample.Tiles.SendTextTile" },
    { "Send tile notification with local images", "SDKSample.Tiles.SendLocalImageTile" },
    { "Send tile notification with web images", "SDKSample.Tiles.SendWebImageTile" },
    { "Send badge notification", "SDKSample.Tiles.SendBadge" },
    { "Send push notifications from a Windows Azure Mobile Service", "SDKSample.Tiles.UsePushNotifications" },
    { "Enable notification queue and tags", "SDKSample.Tiles.EnableNotificationQueue" },
    { "Use notification expiration", "SDKSample.Tiles.NotificationExpiration" },
    { "Image protocols and baseUri", "SDKSample.Tiles.ImageProtocols" },
    { "Globalization, localization, scale, and accessibility", "SDKSample.Tiles.Globalization" },
    { "Content deduplication", "SDKSample.Tiles.ContentDeduplication" }
};


